geocoder_tester/base.py

```diff
@@ -20,6 +22,74 @@
 
 http = requests.Session()
 
+class GenericApi:
+    """ Access proxy for generic geocodejson APIs. The API URL must be
+        the search endpoint.
+
+        This class provides the basic access functions. You should
+        normally choose the specific API type you connect with.
+    """
+
+    def search(self, **params):
+        r = http.get(self.search_url(), params=self.search_params(**params),
+                 headers={'user-agent': 'geocode-tester'})
+        if not r.status_code == 200:
+            raise HttpSearchException(error="Non 200 response")
+        return r.json()
+
+
+    def search_params(self, query, **kwargs):
+        params = {"q": query}
+        if kwargs.get('limit'):
```

Comment:
Can we fall into the trap of passing `limit=0` in some situation ?

Reply:
Photon even accepts `limit=0` and returns (correctly) nothing. I'd still wouldn't expect it to do something sensible.